lib/mix/tasks/openapi.json.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Mix.Tasks.Openapi.Spec.Json do
+  @moduledoc """
+  Serialize the given OpenApi spec module to a json file.
+  """
+  use Mix.Task
+  require Mix.Generator
+
+  @default_filename "openapi.json"
+
+  defmodule Options do
+    @moduledoc false
+
+    defstruct filename: nil, spec: nil, json_codec: nil, endpoint: nil, pretty: false
+  end
+
+  @impl true
+  def run(argv) do
+    opts = parse_options(argv)
+
+    with :ok <- start_endpoint(opts),
+         {:ok, content} <- generate_spec(opts) do
+      write_spec(content, opts.filename)
+    else
+      {:error, error} -> raise error
+    end
+  end
+
+  def start_endpoint(options) do
+    case options.endpoint.start_link() do
+      {:ok, _} ->
+        :ok
+
+      _ ->
+        {:error, "Couldn't start endpoint"}
+    end
+  end
+
+  def generate_spec(options) do
+    case Code.ensure_loaded(options.spec) do
+      {:module, spec} ->
+        if function_exported?(spec, :spec, 0) do
+          {:ok, spec.spec() |> options.json_codec.encode!(pretty: options.pretty)}
+        else
+          {:error, "Spec do not export spec callback"}
+        end
+
+      {:error, :nofile} ->
+        {:error, "Not a valid spec"}
+    end
+  end
+
+  defp parse_options(argv) do
+    parse_options = [strict: [spec: :string, json_codec: :string, pretty: :boolean]]
+    {opts, args, _} = OptionParser.parse(argv, parse_options)
+
+    %Options{
+      filename: args |> List.first() || @default_filename,
+      spec: find_spec(opts),
+      endpoint: endpoint_as_atom(opts),
+      json_codec: json_codec_as_atom(opts),
```

Comment:
You can probably use [OpenApi.json_encoder](https://github.com/open-api-spex/open_api_spex/blob/master/lib/open_api_spex/open_api.ex#L75) for this.